torch/utils/checkpoint.py

```diff
@@ -601,6 +602,20 @@ def _internal_assert(cond):
         )
 
 
+_global_checkpoint_saved_tensor_hooks = OrderedDict()
+
+
+def _register_checkpoint_saved_tensor_hook(hook) -> RemovableHandle:
+    r"""Register a hook to be called when a tensor is saved by checkpoint.
+
+    The hook will be called with the tensor as the only argument.
+    """
```

Comment:
add a comment to emphasize that this is a private API that people in general should not use